File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/MainActivity.java
package expencive.vk.com.dagger2ex;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import javax.inject.Inject;

import expencive.vk.com.dagger2ex.car.Car;
import expencive.vk.com.dagger2ex.dagger.CarComponent;
import expencive.vk.com.dagger2ex.dagger.DaggerCarComponent;
import expencive.vk.com.dagger2ex.dagger.DieselEngineModule;

public class MainActivity extends AppCompatActivity {

    @Inject
    Car car;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        CarComponent component = DaggerCarComponent.builder()
                .dieselEngineModule(new DieselEngineModule(100))
                .build();
        component.inject(this);

        //car = component.getCar();

        car.drive();
    }
}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/MainActivity.kt
package vk.expencive.daggerkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import vk.expencive.daggerkotlin.car.Car
import vk.expencive.daggerkotlin.dagger.CarComponent
import vk.expencive.daggerkotlin.dagger.DaggerCarComponent
import vk.expencive.daggerkotlin.dagger.DieselEngineModule
import javax.inject.Inject

class MainActivity : AppCompatActivity() {
    private val TAG = "MainActivity"

    @Inject lateinit var car: Car

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //val component: CarComponent = DaggerCarComponent.create()
        val component: CarComponent = DaggerCarComponent.builder()
            .dieselEngineModule(DieselEngineModule(100))
            .build()
        component.ingect(this)
        //car = component.getCar()
        car.drive()




    }
}
